### DISCUSSION TITLE
Do you support something like react-native `snapToInterval` option?

### DISCUSSION THREAD

**Discussion by enheit** [pmndrs/react-spring#1693] [Support]
## 🤓 Question

I would like to use `react-spring` and `decay` prop to achieve the natural scrolling behavior. Unfortunately, I can't find how to stop decay animation at the closest row (not at the middle of the row).

![snap](https://user-images.githubusercontent.com/8645216/132422474-05de20d2-d2f0-47e6-92b9-e2221dc5cc6c.gif)

I would like to have some prop like `round` which says `react-spring` where to stop at the end.

Is it possible to achieve this with the current version of the `react-spring`? If yes, how can I do that?

P.S. react-native's ScrollView has prop like [`snapToInterval`](https://reactnative.dev/docs/scrollview#snaptointerval) which stops the view at the closest position to prevent element to be shown partially. 

UPDATE

Also, I was able to find such prop in [popmotion](https://popmotion.io/) lib. They have `modifyTarget` prop which allows to round the end position like `modifyTarget: position => Math.round(position / snapInterval) * snapInterval`

**Comment by raohong** [comment]
`modifyTarget` api is enough, at [spring](https://github.dev/pmndrs/react-spring/blob/68818dc4d29bd7235810af142d903df2f47c9926/packages/core/src/SpringValue.ts#L246-L246)

```
const time = 1 / (1 - decay);
const movement = velocity * time;
const target = modifyTarget( from + movement);
const distance = target - from;
const e = Math.exp(-(1 - decay) * elapsed)

position =  from + target - target * e;
```
